libs/service/api_service.cpp

```diff
@@ -21,47 +21,72 @@ extern "C"
 
 static ebpf_result_t
 _build_helper_id_to_address_map(
-    ebpf_handle_t program_handle, ebpf_code_buffer_t& byte_code, std::map<uint32_t, uint64_t>& helper_id_to_adddress)
+    ebpf_handle_t program_handle, ebpf_code_buffer_t& byte_code, std::vector<uint64_t>& helper_addresses)
 {
+    // Note:
+    // eBPF supports helper IDs in the range [1, MAXUINT32]
+    // uBPF jitter only supports helper IDs in the range [0,63]
+    // Build a table to map [1, MAXUINT32] -> [0,63]
+    std::map<uint32_t, uint32_t> helper_id_mapping;
+
     ebpf_inst* instructions = reinterpret_cast<ebpf_inst*>(byte_code.data());
     for (size_t index = 0; index < byte_code.size() / sizeof(ebpf_inst); index++) {
         ebpf_inst& instruction = instructions[index];
         if (instruction.opcode != INST_OP_CALL) {
             continue;
         }
-        helper_id_to_adddress[instruction.imm] = 0;
+        helper_id_mapping[instruction.imm] = 0;
     }
 
-    if (helper_id_to_adddress.size() == 0)
+    if (helper_id_mapping.size() == 0)
         return EBPF_SUCCESS;
 
+    // uBPF jitter supports a maximum of 64 helper functions
```

Comment:
Do we have a similar limitation in Linux also?

Reply:
I don't think so. The issue comes down to the fact that we have a disjointed helper-id name space, with [0, 0xFFFF] being global and [0x1000, 0x2000] being program type specific. 

uBPF assumes all helper-ids are in the range [0,63].